src/systems/model_photo_shoot/ModelPhotoShoot.cc

```diff
@@ -0,0 +1,283 @@
+void ModelPhotoShoot::Configure(const ignition::gazebo::Entity &_entity,
+                                const std::shared_ptr<const sdf::Element> &_sdf,
+                                ignition::gazebo::EntityComponentManager &_ecm,
+                                ignition::gazebo::EventManager &_eventMgr)
+{
+  std::string save_data_location =
+      _sdf->Get<std::string>("translation_data_file");
+  if (save_data_location.empty())
+  {
+    igndbg << "No data location specified, skipping translaiton data"
+              "saving.\n";
+  }
+  else
+  {
+    igndbg << "Saving translation data to: "
+        << save_data_location << std::endl;
+    this->dataPtr->savingFile.open(save_data_location);
+  }
+
+  if (_sdf->HasElement("random_joints_pose"))
+  {
+    this->dataPtr->randomPoses = _sdf->Get<bool>("random_joints_pose");
+  }
+
+  this->dataPtr->connection =
+      _eventMgr.Connect<ignition::gazebo::events::PostRender>(std::bind(
+          &ModelPhotoShootPrivate::PerformPostRenderingOperations,
+          this->dataPtr.get()));
+
+  this->dataPtr->model = std::make_shared<ignition::gazebo::Model>(_entity);
+  this->dataPtr->modelName = this->dataPtr->model->Name(_ecm);
+  // Get the pose of the model
+  this->dataPtr->modelPose3D =
+      ignition::gazebo::worldPose(this->dataPtr->model->Entity(), _ecm);
+}
+
+//////////////////////////////////////////////////
+void ModelPhotoShoot::PreUpdate(
+    const ignition::gazebo::UpdateInfo &,
+    ignition::gazebo::EntityComponentManager &_ecm)
+{
+  if (this->dataPtr->randomPoses)
+  {
+    std::vector<gazebo::Entity> joints = this->dataPtr->model->Joints(_ecm);
+    unsigned seed =
+        std::chrono::system_clock::now().time_since_epoch().count();
+    std::default_random_engine generator(seed);
+    for (auto joint = joints.begin(); joint != joints.end(); ++joint)
+    {
+      auto jointNameComp = _ecm.Component<components::Name>(*joint);
+      if (jointNameComp && jointNameComp->Data() != "World")
+      {
+        // Using the JointAxis component to extract the joint pose limits
+        auto jointAxisComp = _ecm.Component<components::JointAxis>(*joint);
+        if (jointAxisComp)
+        {
+          float mean =
+              (jointAxisComp->Data().Lower() +
+              jointAxisComp->Data().Upper()) / 2;
+          float stdv = (jointAxisComp->Data().Upper() - mean) / 3;
+          std::normal_distribution<float> distribution(mean, stdv);
+          float jointPose = distribution(generator);
+          _ecm.SetComponentData<components::JointPositionReset>(*joint,
+                                                                {jointPose});
+          if (this->dataPtr->savingFile.is_open())
+          {
+            this->dataPtr->savingFile << jointNameComp->Data() << ": "
+                                      << jointPose << std::endl;
+          }
+        }
+        else
+        {
+          igndbg << "No jointAxisComp found, ignoring joint: " <<
```

Comment:
Should this be at least a warning, if not an error?

Reply:
I wasn't sure if there was any valid case in which a `joint` could not have a `jointAxisComp`. I'll make it an error then.